frontend/catalyst/compilation_pipelines.py

```diff
@@ -459,8 +459,9 @@ class QJIT:
     """
 
     def __init__(self, fn, compile_options):
-        self.compiler = Compiler()
         self.compile_options = compile_options
+        self.compiler = Compiler(compile_options)
+        self.compiling_from_textual_ir = isinstance(fn, str)
```

Comment:
> When does this compiling from string use?

When we want to write MLIR and feed it to the QJIT option.

> Is this a development/debugging option?

Yes.

> Do we parse this IR in Python to check the validity?

The IR is parsed but it has never been parsed in Python, otherwise we would need to write our own parser for MLIR. Instead the IR is parsed from C++.

> and get the function name before compiling?

The function name when compiling from IR is set to be `catalyst.entry_point`. Otherwise, we don't know which function is the entry point. There are several ways of doing this, but this one is good enough.